lib/ae_mdw_web/websocket/subscriptions.ex

```diff
@@ -48,41 +50,43 @@ defmodule AeMdwWeb.Websocket.Subscriptions do
     {:ok, :no_state}
   end
 
-  @spec subscribe(pid(), channel()) ::
+  @spec subscribe(pid(), version(), channel()) ::
           {:ok, [channel()]} | {:error, :invalid_channel | :already_subscribed}
-  def subscribe(pid, channel) do
-    with {:ok, channel_key} <- validate_subscribe(pid, channel) do
+  def subscribe(pid, version, channel) do
+    with {:ok, channel_key} <- validate_subscribe(pid, version, channel) do
       maybe_monitor(pid)
-      :ets.insert(@subs_pid_channel, {pid, channel})
+      :ets.insert(@subs_pid_channel, {pid, {channel_key, channel}})
       :ets.insert(@subs_channel_pid, {channel_key, pid})
 
       {:ok, subscribed_channels(pid)}
     end
   end
 
-  @spec unsubscribe(pid(), channel()) ::
+  @spec unsubscribe(pid(), version(), channel()) ::
           {:ok, [channel()]} | {:error, :invalid_channel | :not_subscribed}
-  def unsubscribe(pid, channel) do
-    with {:ok, channel_key} <- validate_unsubscribe(pid, channel) do
+  def unsubscribe(pid, version, channel) do
+    with {:ok, channel_key, channel} <- validate_unsubscribe(pid, version, channel) do
       :ets.delete_object(@subs_channel_pid, {channel_key, pid})
-      :ets.delete_object(@subs_pid_channel, {pid, channel})
+      :ets.delete_object(@subs_pid_channel, {pid, {channel_key, channel}})
```

Comment:
In `subscribed_channels` the return of `@subs_pid_channel` it would have a different format (includes the version):
`[{{_version, _channel_key}, channel}]`

Seems a good moment for extending the test cases to cover it.

Reply:
The problem is the variable down below is named incorrectly, it should have been `{_channel_key, channel}` instead, since we don't really need to pattern match on the `{_version, _channel}` tuple. Changing it

Reply:
Besides the naming the number of elements returned as well because what is returned is the value of the record without the key:
```
    @subs_pid_channel
    |> :ets.match({pid, :"$1"})
```
`[{_channel_key, channel}]` instead of `[{{_version, _channel_key}, channel}]` and since the channel key can be easily derived wouldn't store it but it's ok if you really want to keep in memory for `unsubscribe_all`.

Reply:
I see now the number of elements is ok, the confusion is because the version is part of channel key and in this `Enum.map` they are separate as `[{{_version, _channel_key}, channel}]`